src/operation.jl

```diff
@@ -59,6 +59,11 @@ for FUN in (:applyeager, :applylazy, :applypermute,
         @inline function ($FUN)(op::Operation, img::AbstractArray)
             ($FUN)(op, img, randparam(op, img))
         end
+
+        # Semantic wrapper support
+        @inline $FUN(op::Operation, img::SemanticWrapper, param) = $FUN(op, img, param, shouldapply(op, img))
+        @inline $FUN(op::Operation, img::SemanticWrapper, param, ::Val{false}) = img
+        @inline $FUN(op::Operation, img::Mask, param, ::Val{true}) = Mask(($FUN)(op, unwrap(img), param))
```

Comment:
A nice fix 👍

Reply:
I was also thinking about changing
```julia
@inline $FUN(op::Operation, img::Mask, param, ::Val{true}) = Mask(($FUN)(op, unwrap(img), param))
```
to something like
```julia
$FUN(op::Operation, img::T, param, ::Val{true}) where {T<:SemanticWrapper} = T(($FUN)(op, unwrap(img), param))
```
Then all semantic wrappers would have to define an appropriate constructor besides `unwrap`. What do you think?

Reply:
You clearly know this design better than me :)

Reply:
I just noticed that our Mask (and semantic wrappers in general) behave like a functor. There is a package [Functors.jl](https://fluxml.ai/Functors.jl/), which I am not too familiar with currently but I believe it would allow us to remove all the unwrapping-and-wrapping, e.g.,
```julia
#currently
@inline contiguous(A::T) where {T <: SemanticWrapper} =T(contiguous(unwrap(A)))
#as functor
@inline contiguous(A::SemanticWrapper) = fmap(contiguous, A)

#currently
@inline _plain_array(A::T) where {T <: SemanticWrapper} =T(_plain_array(unwrap(A)))
#as functor
@inline _plain_array(A::SemanticWrapper) = fmap(_plain_array, A)

#currently
$FUN(op::Operation, img::T, param, ::Val{true}) where {T<:SemanticWrapper} = T(($FUN)(op, unwrap(img), param))
#as functor
$FUN(op::Operation, sw::T, param, ::Val{true}) where {T<:SemanticWrapper} = fmap(img -> ($FUN)(op, img, param), sw)
```

Might be worth the extra dependency.

Reply:
The more "magic" and foreign features/macros you introduce into the codebase, the less maintainable it is.

Functors are designed specifically to extract and pass information for nested data structures. Networks are a typical use case. But in our case, it's just one layer `Mask` here and I don't think it's necessary at the moment.

Reply:
I am sorry, I need help. I am trying to do
```julia
$FUN(op::Operation, img::T, param, ::Val{true}) where {T<:SemanticWrapper} = T(($FUN)(op, unwrap(img), param))
```
which fails because `($FUN)(op, unwrap(img), param)` may return a different type than what `img` is, which causes an error in the construction `T(...)` as `T == Augmentor.Mask{typeof(img)}`. Is there a way around it?

(In the current code, even `T(img)` fails because it lacks `Mask{T}(img:T) where {T<:AbstractArray}` constructor)